src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -543,12 +543,17 @@ def get_login_credentials(self, resource=None, subscription_id=None, aux_subscri
         identity_type, identity_id = Profile._try_parse_msi_account_name(account)
 
         external_tenants_info = []
-        ext_subs = [aux_sub for aux_sub in (aux_subscriptions or []) if aux_sub != subscription_id]
-        for ext_sub in ext_subs:
-            sub = self.get_subscription(ext_sub)
-            if sub[_TENANT_ID] != account[_TENANT_ID]:
-                # external_tenants_info.append((sub[_USER_ENTITY][_USER_NAME], sub[_TENANT_ID]))
-                external_tenants_info.append(sub)
+        if isinstance(aux_tenants, list):
+            external_tenants_info = [tenant for tenant in aux_tenants if tenant != account[_TENANT_ID]]
+            if aux_subscriptions is not None:
+                logger.warning("When aux-subscriptions and aux-tenants pass in at the same time, "
+                               "only aux-tenants will work")
```

Comment:
I'd prefer to error out when both are specified, as giving warning instead encourages incorrect usage. 

https://github.com/Azure/azure-cli/blob/5edbd822ee860d7ba5d5182811b68233e0846ba8/src/azure-cli-core/azure/cli/core/_profile.py#L605

Reply:
It has been verified in the validator and an error has been returned for this scenario.
This method is a underlying method, which is also called in many other places. Just logging warning will have a smaller impact on different businesses, unless we can confirm that all scenarios calling it in the future can directly report an error.

Reply:
In other places, no invocation involves `aux_tenants`. Making them mutual-exclusive won't cause any breaking change.

Reply:
Yes, I know it has no effect on the existing calling methods. I mean, I didn't know if future calls should be waring or error.
I think you're right. I'll change the logic.